packages/backend/src/helpers/get-test-execution-steps.ts

```diff
@@ -19,8 +19,37 @@ export async function getTestExecutionSteps(
     })
     .throwIfNotFound()
 
+  if (!flow.steps.length) {
+    return []
+  }
+
   if (flow.testExecution && !ignoreTestExecutionId) {
-    return flow.testExecution.executionSteps
+    const testExecutionSteps = flow.testExecution.executionSteps
+    testExecutionSteps.sort((a, b) => a.step.position - b.step.position)
+
+    /**
+     * Sanity check to ensure not more than 1 execution step per step is returned
+     * If more than 1 exists, we return the latest one sorted by createdAt
+     */
+    const stepIds = new Set<string>()
+    const dedupedTestExecutionSteps = testExecutionSteps.reduce((acc, curr) => {
+      if (stepIds.has(curr.stepId)) {
+        const otherExecutionStep = acc[acc.length - 1]
+        // possible bug in single step testing !! this should not happen
```

Comment:
as long as you delete the execution step everytime you test step, this should not occur right?

Reply:
yes but just in case, that's why i console.warn if this happens